src/azul/chalice.py

```diff
@@ -14,25 +14,25 @@
     config,
 )
 from azul.json import json_head
-from azul.openapi import (
-    clean_specs,
-    merge_dicts,
-    join_specs,
-)
 from azul.types import (
     LambdaContext,
     JSON,
+    MutableJSON,
 )
 
 log = logging.getLogger(__name__)
 
 
 class AzulChaliceApp(Chalice):
 
-    def __init__(self, app_name, unit_test=False):
+    def __init__(self, app_name, unit_test=False, toplevel_spec=None):
         self.unit_test = unit_test
-        self.path_specs: Dict[str, JSON] = {}
-        self.method_specs: Dict[Tuple[str, str], JSON] = {}
+        if toplevel_spec is not None:
+            assert 'paths' not in toplevel_spec, 'Toplevel spec should not specify paths'
+            self.specs: Optional[MutableJSON] = cast(MutableJSON, copy.deepcopy(toplevel_spec))
```

Comment:
This fragment occurs often enough to warrant a helper function. Do a search for `cast` and `deepcopy` across the codebase. Signature should be `def copy_json(o:JSON)->MutableJSON` and it should probably live in the `azul.json` module.

Reply:
I found another instance where both `cast` and `deepcopy` are used, but the cast is not to JSON, so I think I'll leave it as is https://github.com/DataBiosphere/azul/blob/develop/src/azul/service/manifest_service.py#L720.